pkg/exclusions/exclusions_test.go

```diff
@@ -0,0 +1,303 @@
+package exclusions
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/osv-scanner/internal/sbom"
+	"github.com/google/osv-scanner/pkg/lockfile"
+	"github.com/google/osv-scanner/pkg/models"
+)
+
+type MockReporter struct {
+	Output string
+	Error  string
+}
+
+func (r *MockReporter) PrintText(text string) {
+	r.Output = text
+}
+func (r *MockReporter) PrintError(text string) {
+	r.Error = text
+}
+
+func (r *MockReporter) HasPrintedError() bool {
+	return false
+}
+func (r *MockReporter) PrintResult(vulnResult *models.VulnerabilityResults) error {
+	return nil
+}
+
+func TestExcludePackages(t *testing.T) {
+	t.Parallel()
+	type testCase struct {
+		name                 string
+		lockfileData         []lockfile.PackageDetails
+		exclusionPatterns    []string
+		expectedLockfileData []lockfile.PackageDetails
+		wantErr              bool
+	}
```

Comment:
You can then had to the testCase 2 new checks:
- `OutputText`
- `ErrorText`

Verify both `OutputText` and `ErrorText` match from the MockReporter.